sparrow-py/pysrc/sparrow_py/session.py

```diff
@@ -2,17 +2,52 @@
 from sparrow_py import Expr
 from typing import Optional
 import pyarrow as pa
+import pandas as pd
+
+class InvalidSchemaException(Exception):
```

Comment:
We should decide if we want to define new exceptions for things, or use existing exceptions. I kind of like having more specific ones, but so far I've been trying to use existing things (eg., `ValueError` or `TypeError`, etc.)

Reply:
Based on this: https://www.perplexity.ai/search/9d6868a2-8c13-4bae-93ed-a21c5af25af5?s=u

I think we should probably plan to define our own. This looks reasonable for now, although I think we should probably call this an `IncompatibleSchemaError` or something like that (eg., the schema is valid, but isn't compatible with our expectations) and it should end in `Error` like most of the Python exceptions.

In the future, this exception may be defined in Rust and re-thrown, but I'm not sure.